Assets/StreamingAssets/LUA/Furniture.lua

```diff
@@ -560,6 +560,37 @@ function AirPump_GetSpriteName(furniture)
     return furniture.Type .. suffix
 end
 
+function Vent_OnUpdate(furniture, deltaTime)
+    furniture.SetAnimationProgressValue(furniture.Parameters["openness"].ToFloat(), 1)
+    furniture.Tile.EqualiseGas(deltaTime * furniture.Parameters["gas_throughput"].ToFloat() * furniture.Parameters["openness"].ToInt())
+end
+
+function Vent_SetOrientationState(furniture)
+    local t = furniture.Tile
+    if (furniture.Tile == nil) then
+        return furniture.Type
+    end
+    
+    local north = World.Current.GetTileAt(t.X, t.Y + 1, t.Z)
```

Comment:
Use tile.North() ... tile.South()  ... tile.West() ... tile.East() ... You might not even need the variables.